src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -33,22 +35,23 @@ public RegionDiscoveryProvider(IHttpManager httpManager, INetworkCacheMetadataPr
 
         public async Task<InstanceDiscoveryMetadataEntry> GetMetadataAsync(Uri authority, RequestContext requestContext)
         {
-            var logger = requestContext.Logger;
-
+            ICoreLogger logger = requestContext.Logger;
             string environment = authority.Host;
-            var cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
-            if (cachedEntry != null)
+            InstanceDiscoveryMetadataEntry cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
+            
+            if (cachedEntry == null)
+            {
+                Uri regionalizedAuthority = await BuildAuthorityWithRegionAsync(authority, requestContext.Logger).ConfigureAwait(false);
+                CacheInstanceDiscoveryMetadata(CreateEntry(authority, regionalizedAuthority));
+
+                cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
+                logger.Verbose($"[Region Discovery] Created metadata for the regional environment {environment} ? {cachedEntry != null}");
+            } else
             {
                 logger.Verbose($"[Region Discovery] The network provider found an entry for {environment}");
-                return cachedEntry;
             }
 
-            Uri regionalizedAuthority = await BuildAuthorityWithRegionAsync(authority, requestContext.Logger).ConfigureAwait(false);
-            CacheInstanceDiscoveryMetadata(CreateEntry(authority, regionalizedAuthority));
-
-            cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
-            logger.Verbose($"[Region Discovery] Created an entry for the regional environment {environment} ? {cachedEntry != null}");
-
+            requestContext.ApiEvent.RegionDiscovered = cachedEntry.PreferredNetwork.Split('.')[0];
```

Comment:
>requestContext.ApiEvent.RegionDiscovered = cachedEntry.PreferredNetwork.Split('.')[0]; [](start = 12, length = 86)

should this have been parsed with the uri component?

Reply:
Then it won't get logged when the instance is cached. So, I added it here.